src/utils/chunkAssignment.ts

```diff
@@ -285,9 +285,12 @@ function analyzeModuleGraph(entries: Iterable<Module>): {
 					dynamicEntryModules.add(resolution);
 					allEntriesSet.add(resolution);
 					dynamicImportsForCurrentEntry.add(resolution);
-					if (resolution.includedDirectTopLevelAwaitingDynamicImporters.has(currentEntry)) {
-						awaitedDynamicEntryModules.add(resolution);
-						awaitedDynamicImportsForCurrentEntry.add(resolution);
+					for (const includedDirectTopLevelAwaitingDynamicImporter of resolution.includedDirectTopLevelAwaitingDynamicImporters) {
+						if (modulesToHandle.has(includedDirectTopLevelAwaitingDynamicImporter)) {
+							awaitedDynamicEntryModules.add(resolution);
+							awaitedDynamicImportsForCurrentEntry.add(resolution);
+							break;
+						}
```

Comment:
Looks good to me, thanks! I needed to think about this for a moment, maybe we want to rename `modulesToHandle` to something like `staticDependencies`, that would clarify that this is not just a transient Set but really has a meaning. Otherwise, this looks fine to me.

Reply:
Done!